File: fhg-api/src/main/java/de/jbamberger/api/FhgApiModule.java
package de.jbamberger.api;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
public abstract class FhgApiModule {

    @Binds
    @Singleton
    public abstract FhgApi bindsFhgApi(FhgApiImpl impl);
}

File: fhg-api/src/main/java/de/jbamberger/api/FhgApiModule.kt
package de.jbamberger.api

import dagger.Binds
import dagger.Module
import javax.inject.Singleton

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module(includes = arrayOf(NetModule::class))
abstract class FhgApiModule {

    @Binds
    @Singleton
    internal abstract fun bindsFhgApi(impl: FhgApiImpl): FhgApi
}